# Human resources in science and technologies
 
 partial reproduction of the [Statistics Explained](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Human_resources_in_science_and_technology&oldid=395960) article

In [1]:
datum<-Sys.Date()

Last modification: {{datum}}

In [2]:
# CONFIGURATION VALUES

year<-2020
age_group1<-'25 to 64' 
age_group2<-'15 to 74'


In [3]:
# CALCULATION
library(restatapi)
eu_cc<-get("cc",envir=.restatapi_env)$EU27_2020
# get_eurostat_dsd("hrst_st_nocc")
value0<-get_eurostat_data("hrst_st_nocc",date_filter=year,filters=c("HRSTO$","EU27_2020","THS","TOTAL",gsub(" to ","-",age_group1)),exact_match=F)$values
value1<-get_eurostat_data("hrst_st_nocc",date_filter=year,filters=c("HRSTO$","EU27_2020","THS","TOTAL",gsub(" to ","-",age_group2)),exact_match=F)$values
value2<-(get_eurostat_data("hrst_st_nocc",date_filter=year,filters=c("HRSTO$","EU27_2020","THS","TOTAL",gsub(" to ","-",age_group2)),exact_match=F)$values/get_eurostat_data("hrst_st_nocc",date_filter=(year-1),filters=c("HRSTO$","EU27_2020","THS","TOTAL",gsub(" to ","-",age_group2)),exact_match=F)$values-1)*100
value2_text<-if (value2>0) {'increase'} else {'decrease'}
value3<-(get_eurostat_data("hrst_st_nocc",date_filter=year,filters=c("HRSTO$","EU27_2020","THS","TOTAL",gsub(" to ","-",age_group1)),exact_match=F)$values/get_eurostat_data("hrst_st_nocc",date_filter=(year-1),filters=c("HRSTO$","EU27_2020","THS","TOTAL",gsub(" to ","-",age_group1)),exact_match=F)$values-1)*100
value3_text<-if (value3>0) {'increase'} else {'decrease'}
value4<-get_eurostat_data("hrst_st_nocc",date_filter=year,filters=c("HRSTO$","EU27_2020","^prof","^perc",gsub(" to ","-",age_group1)),exact_match=F,ignore.case=T)$values
value5<-get_eurostat_data("hrst_st_nocc",date_filter=year,filters=c("HRSTO$","EU27_2020","^tech","^perc",gsub(" to ","-",age_group1)),exact_match=F,ignore.case=T)$values


restatapi: - config file with the API version 1 loaded from GitHub (the 'current' API version number is 1).
           - 4 from the 64 cores are used for parallel computing, can be changed with 'options(restatapi_cores=...)'
           - 'auto' method will be used for file download, can be changed with 'options(restatapi_dmethod=...)'
           - the Table of contents (TOC) was not pre-loaded into the deafult cache ('.restatapi_env').



In [5]:
dsd<-get_eurostat_dsd("hrst_st_nocc")
#dsd
#create_filter_table(list(geo=eu_cc,isco="oc2",category="hrsto",unit="pc",age=gsub(" to ","-",age_group1)),F,dsd,exact_match=F,name=T)
get_eurostat_data("hrst_st_nocc",date_filter=year,filters=list(geo=eu_cc,isco08="oc2",category="hrsto",unit="pc",age=gsub(" to ","-",age_group1)),exact_match=F,name=F,ignore.case=T,label=T,verbose=F)


pattern concept code   name                                            
1  BE      GEO     BE     Belgium                                         
2  BG      GEO     BG     Bulgaria                                        
3  CZ      GEO     CZ     Czechia                                         
4  DK      GEO     DK     Denmark                                         
5  DE      GEO     DE     Germany (until 1990 former territory of the FRG)
6  EE      GEO     EE     Estonia                                         
7  IE      GEO     IE     Ireland                                         
8  EL      GEO     EL     Greece                                          
9  ES      GEO     ES     Spain                                           
10 FR      GEO     FR     France                                          
11 FR      GEO     DE     Germany (until 1990 former territory of the FRG)
12 HR      GEO     HR     Croatia                                         
13 IT      GEO     IT     Italy                                           
14 CY      GEO     CY     Cyprus                                          
15 LV      GEO     LV     Latvia                                          
16 LT      GEO     LT     Lithuania                                       
17 LU      GEO     LU     Luxembourg                                      
18 HU      GEO     HU     Hungary                                         
19 MT      GEO     MT     Malta                                           
20 NL      GEO     NL     Netherlands                                     
21 AT      GEO     AT     Austria                                         
22 PL      GEO     PL     Poland                                          
23 PT      GEO     PT     Portugal                                        
24 RO      GEO     RO     Romania                                         
25 SI      GEO     SI     Slovenia                                        
26 SK      GEO     SK     Slovakia                                        
27 FI      GEO     FI     Finland                                         
28 SE      GEO     SE     Sweden                                          
29 25-64   AGE     Y25-64 From 25 to 64 years

unit                           category                                  
1  Percentage of total employment Persons employed in science and technology
2  Percentage of total employment Persons employed in science and technology
3  Percentage of total employment Persons employed in science and technology
4  Percentage of total employment Persons employed in science and technology
5  Percentage of total employment Persons employed in science and technology
6  Percentage of total employment Persons employed in science and technology
7  Percentage of total employment Persons employed in science and technology
8  Percentage of total employment Persons employed in science and technology
9  Percentage of total employment Persons employed in science and technology
10 Percentage of total employment Persons employed in science and technology
11 Percentage of total employment Persons employed in science and technology
12 Percentage of total employment Persons employed in science and technology
13 Percentage of total employment Persons employed in science and technology
14 Percentage of total employment Persons employed in science and technology
15 Percentage of total employment Persons employed in science and technology
16 Percentage of total employment Persons employed in science and technology
17 Percentage of total employment Persons employed in science and technology
18 Percentage of total employment Persons employed in science and technology
19 Percentage of total employment Persons employed in science and technology
20 Percentage of total employment Persons employed in science and technology
21 Percentage of total employment Persons employed in science and technology
22 Percentage of total employment Persons employed in science and technology
23 Percentage of total employment Persons employed in science and technology
24 Percentage of total employment Persons employed in science and technology
25 Percentage of total employment Persons employed in science and technology
26 Percentage of total employment Persons employed in science and technology
27 Percentage of total employment Persons employed in science and technology
   isco08        age                
1  Professionals From 25 to 64 years
2  Professionals From 25 to 64 years
3  Professionals From 25 to 64 years
4  Professionals From 25 to 64 years
5  Professionals From 25 to 64 years
6  Professionals From 25 to 64 years
7  Professionals From 25 to 64 years
8  Professionals From 25 to 64 years
9  Professionals From 25 to 64 years
10 Professionals From 25 to 64 years
11 Professionals From 25 to 64 years
12 Professionals From 25 to 64 years
13 Professionals From 25 to 64 years
14 Professionals From 25 to 64 years
15 Professionals From 25 to 64 years
16 Professionals From 25 to 64 years
17 Professionals From 25 to 64 years
18 Professionals From 25 to 64 years
19 Professionals From 25 to 64 years
20 Professionals From 25 to 64 years
21 Professionals From 25 to 64 years
22 Professionals From 25 to 64 years
23 Professionals From 25 to 64 years
24 Professionals From 25 to 64 years
25 Professionals From 25 to 64 years
26 Professionals From 25 to 64 years
27 Professionals From 25 to 64 years
   geo                                              time values
1  Austria                                          2020 54.8  
2  Belgium                                          2020 65.1  
3  Bulgaria                                         2020 65.8  
4  Cyprus                                           2020 61.2  
5  Czechia                                          2020 49.2  
6  Germany (until 1990 former territory of the FRG) 2020 51.5  
7  Denmark                                          2020 62.7  
8  Estonia                                          2020 59.3  
9  Greece                                           2020 70.2  
10 Spain                                            2020 62.2  
11 Finland                                          2020 60.8  
12 France                                  

### markdown version

**Tweet:** 
In {{year}}, over ~~67~~ {{round(value0/1000)}} million people aged from {{message(age_group1)}} were employed in science and technology (HRSTO) in the EU. 

This article provides information on statistics [on human resources in science and technology (HRST)](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:Human_resources_in_science_and_technology_(HRST)) which help us to better understand the demand for and the supply of people with strong qualifications in science and technology. It describes the current stock of HRST in the [European Union](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:European_Union_(EU)), the [EFTA](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:European_Free_Trade_Association_(EFTA)) countries and the [candidate countries](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:Candidate_countries) and gives information on the current and future supply of highly skilled people from universities and other specialised education institutions in the HRST field.

Investment in research, development, education and skills is a key policy area for the EU as it is essential for economic growth and for the development of a [knowledge-based economy](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:Knowledge-based_economy). There is, therefore, a great need to measure and analyse the most highly skilled part of the [labour force](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:Labour_force), both within the EU and internationally. 

### Professionals and technicians employed in science and technology occupations

In {{year}}, ~~72.9~~ {{round(value1/1000,1)}} million people in the EU aged from {{message(age_group2)}} were employed in science and technology (and therefore considered as human resources in science and technology by occupation — HRSTO (see Figure 5 below)), an {{ message(value2_text) }} of ~~1.7~~ {{abs(round(value2,1))}} % compared with {{year-1}}.

Within this ~~72.9~~ {{round(value1/1000,1)}} million, ~~67.7~~ {{round(value0/1000,1)}} million people were in the {{message(age_group1)}} age group, an {{ message(value3_text) }} of ~~1.9~~ {{abs(round(value3,1))}} % compared with {{year-1}}. From this broad group ~~57.0~~ {{value4}} % were ['professionals'](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:Professionals_group) (22.6% of ‘Scientists and Engineers’ and 34.4% of ‘Other professionals (other than SE)’) and ~~43.0~~ {{value5}} % were [‘technicians’](https://ec.europa.eu/eurostat/statistics-explained/index.php?title=Glossary:Technicians_and_associate_professionals_group). However, the percentage differed greatly between Member States. Luxembourg (72.3 %), Lithuania (72.1 %) and Romania (70.8 %) reported the highest proportion of professionals. Other Member States with more than two thirds professionals were Greece (70.2 %), Ireland (69.0 %) and Portugal (67.0 %) (see Figure 1). 